pennylane/vqe/vqe.py

```diff
@@ -370,70 +370,145 @@ class VQECost:
             Supports all interfaces supported by the :func:`~.qnode` decorator.
         diff_method (str, None): The method of differentiation to use with the created cost function.
             Supports all differentiation methods supported by the :func:`~.qnode` decorator.
+        optimize (bool): Whether to optimize the observables composing the Hamiltonian by
+            separating them into qubit-wise commuting groups. Each group can then be executed
+            within a single QNode, resulting in fewer QNodes to evaluate.
 
     Returns:
         callable: a cost function with signature ``cost_fn(params, **kwargs)`` that evaluates
         the expectation of the Hamiltonian on the provided device(s)
```

Comment:
Just a comment: in general, we use ``VQECost`` to evaluate the expectation value of any observable, no only the Hamiltonian.
Maybe:
```suggestion
        the expectation of the observable on the provided device(s)
```
Not 100% sure of the renaming since VQE is very tightly associated with finding the lowest-energy state of the Hamiltonian.

Reply:
> e.g., total spin still gets mapped to a qubit Hamiltonian (?)

All many-body observables, including the total spin, are mapped to qubit operators.